### ISSUE TITLE
Simple Query, Firebird 100%CPU client unresponsive

### ISSUE THREAD

**Issue by DaltonCalford** [aafemt/magpie#1]
Base query is a select count(*) on an empty table

ie select count(*)  from sto_actions.
it works in isql (odbc sql)

the client is fb_isql (firebird command line sql)
set term ^^;
execute block returns (id bigint) as
begin 
  for execute statement
    'select count(*) from sto_actions'
    on external 'odbc://AM-MARIADB10PROD'
    into :id
  do suspend;
end ^^

**/etc/odbcinst.ini**
[MariaDB ODBC 3.1 Driver]
Description=MariaDB Connector/ODBC v.3.1
Driver=/usr/lib/libmaodbc.so
UsageCount=2

**DSN**
[AM-MARIADB10PROD]
Description=Production MariaDB System
Driver=MariaDB ODBC 3.1 Driver
SERVER=AM-MARIADB10
PORT=3306
USER={USER}
PASSWORD={PASSWORD}
DATABASE=PRODUCTION

**output of magpie log**
x7effdc012850  ODBCProvider created
0x7effdc012850  ODBCProvider::setDbCryptCallback(...)
0x7effdc012850  ODBCProvider::attachDatabase("/opt/firebird/security5.fdb")
0x7effdc012850  ODBCProvider::release
0x7effdc012850  ODBCProvider destructed (counter 0)
0x7effeda342a0  ODBCFactory::createPlugin(0x7effee55fbf8)
Unicode charset is UTF16LE
0x7effeda342a0          Default config = 0x7effee55fbb8
                License string "JwtS1UJzq6geC5jzytU1oP0fGXr0eDkSKTH5tmN58vuhamE0JqlgRwbGPwliI+CjRE0DIlNRy373qt/3Hoa5k5cfEAqLMrEz9IWAaiu/2qruV6GJp9TZur3HRDv2VAupotzbEDDPMs6DKFW51H4ihbeNhjnsDuulDBAWGHf9RIOU"
0x7effdc012850  ODBCProvider created
0x7effdc012850  ODBCProvider::setDbCryptCallback(...)
0x7effdc012850  ODBCProvider::attachDatabase("odbc://AM-MARIADB10PROD")
DPB version 1:
        isc_dpb_ext_call_depth 1
        isc_dpb_ext_call_depth 1
        isc_dpb_auth_block
                Auth block 0, length 38
                        AUTH_TYPE: Seen
                        AUTH_SECURE_DB: /home/databases/olap.fdb
                Auth block 1, length 41
                        AUTH_TYPE: Seen
                        AUTH_SECURE_DB: /opt/firebird/security5.fdb
                Auth block 2, length 63
                        AUTH_TYPE: USER
                        AUTH_NAME: SYSDBA
                        AUTH_PLUGIN: Srp256
                        AUTH_SECURE_DB: /opt/firebird/security5.fdb
        isc_dpb_lc_ctype UTF8

**Comment by reevespaul** [comment]
I have a similar DSN and I used this sql

```sql
 for execute statement
    'select count(*) from  wp_posts'
  on external 'odbc://wp'
  into :id
```
and got this error immediately. No system overload at all.

```
Statement failed, SQLSTATE = 42000
Execute statement error at attach :
335544421 : connection rejected by remote interface
140635776340560 : [unixODBC][ma-3.1.10]Access denied for user 'firebird'@'localhost' (using password: NO)
Data source : Firebird::odbc://wp
-At block line: 3, col: 3
After line 1 in file /home/paul/Documents/IBP/magpie/magpie-example-count-mysql.dml
```

AFAICT the only difference is my host is running opensuse 15.5

You could try adding the un/pw to the sql like this:

```sql
  for execute statement
    'select count(*) from  wp_posts'
  as user 'wordpressuser'
  password 'letmein'
  on external 'odbc://wp'
  into :id
```

and you should get something like this:

```
 isql -i /home/paul/Documents/IBP/magpie/magpie-example-count-mysql.dml localhost:employee

                   ID 
===================== 
                  429 

```

I'm not happy that the mysql driver wants the user name and password embedded in the sql. We had this problem yesterday with the postgres driver but we fixed that so I'm a bit disappointed that the problem exists still with mysql. More work needed there, I guess.

Anyway, let us know if adding the un/pw to the sql 'fixes' the problem.

And what is the exact version of the driver ? I'm currently using 3.1.10

**Comment by DaltonCalford** [comment]
Same results with user name/password.
Updated to latest maria connector. Same results.

The mariadb connector is version 3.2.3

This is the code with the changes
execute block returns (id bigint) as
begin
 for execute statement
  'select count(*) from sto_actions'
  on external 'odbc://AM-MARIADB10PROD'
  as user '****'
  password '****'
  into :id
  DO suspend;
end

**Comment by reevespaul** [comment]
Can you try connecting to this public postgres server

```ini
[rna]
# See https://rnacentral.org/help/public-database
Driver               = PSQL
Description          = "Test odbc connection to the RNA database."
UserName             = reader
Password             = NWDMCE5xdipIjRrp
ServerName           = hh-pgsql-public.ebi.ac.uk
Port                 = 5432
Database             = pfmegrnargs
ByteaAsLongVarBinary = 1
```

with this sql:

```sql
set term ^;
execute block returns (upi varchar(13), taxid varchar(4), ac varchar(64)) as
begin
  for execute statement
    'select  upi,   taxid,  ac from xref where ac in (''OTTHUMT00000106564.1'', ''OTTHUMT00000416802.1'')'
    on external 'odbc://rna'
    into :upi, :taxid, :ac
  do
    suspend;
end ^
set term ;^
```

It takes a moment to run but you should get two rows back.

**Comment by aafemt** [comment]
> output of magpie log

Is it whole log?

**Comment by aafemt** [comment]
@DaltonCalford
This issue should be fixed in [release 1.0.0.2](https://github.com/aafemt/magpie/releases/tag/v1.0.0.2). Check, please.
